packages/cli/src/commands/setup/ui/libraries/chakra-ui.js

```diff
@@ -51,22 +59,42 @@ export async function handler({ force, install }) {
     },
     {
       title: 'Setting up Chakra UI...',
-      skip: () => checkSetupStatus() === 'done',
-      task: () => wrapWithChakraProvider(),
+      skip: () => fileIncludes(rwPaths.web.app, 'ChakraProvider'),
+      task: () =>
+        extendJSXFile(rwPaths.web.app, {
+          insertComponent: {
+            name: 'ChakraProvider',
+            props: { theme: 'extendedTheme' },
+            within: 'RedwoodProvider',
+            insertBefore: '<ColorModeScript />',
```

Comment:
Writing this comment to remind myself to bring this up with Dom when we review the PR. This is just an idea. We don't have to act on it.

I need to look into the `extendJSXFile` function below to see how this works. But just looking at it here it seems as if this will insert two components, `<ChakraProvider>` and `<ColorModeScript />`.

I wonder if it would be easier to learn, read and understand if it always just inserted one component (otherwise maybe we should consider renaming the method to `insertComponents`).

So maybe something like this

```
insertComponent: {
  name: 'ChakraProvider',
  props: { theme: 'extendedTheme' },
  within: 'RedwoodProvider',
}

// ...

insertComponent: {
  name: 'ColorModeScript',
  before: 'ChakraProvider',
}
```

If you use the `within: 'ExistingComponent` option it would create `<Component>` and `</Component>` wrapping all current children inside `<ExistingComponent>`

If you use the `before: ExistingComponent` option it would create `<Component />` and inserting it before `<ExistingComponent>` or `<ExistingComponent />`